tests/integration/ua_emulation.rs

```diff
@@ -303,22 +303,19 @@ async fn test_ua_emulation() {
         )
             .into_layer(service_fn(async |mut req: Request| {
                 // We can edit our current builder directly or create a new one if needed
-                let builder = match req.extensions_mut().get_mut::<TlsConnectorDataBuilder>() {
-                    Some(builder) => builder,
-                    None => req
-                        .extensions_mut()
-                        .insert_mut(TlsConnectorDataBuilder::default()),
-                };
+                let mut builder = req
+                    .extensions_mut()
+                    .get::<TlsConnectorDataBuilder>()
+                    .cloned()
+                    .unwrap_or_default();
                 builder.set_server_verify_mode(ServerVerifyMode::Disable);
+                req.extensions_mut().insert(builder);
```

Comment:
Why do you insert the builder again, if it already comes from the req??? I mean you clone it just to modify it and put it back??? What am I missing here, would you not just either get it or make a new one and inert, but in case exist just modify the existing one?